Version 1.8.1 no longer compatible with Kotlin < 1.8

It seems that version 1.8.1 lost compatibility with Kotlin < 1.8.

I use `notion-sdk-jvm` modules in [`spring-batch-notion`](https://github.com/scordio/spring-batch-notion) and I [cross-build](https://github.com/scordio/spring-batch-notion/blob/61794c38f2c7f28aa67fdd1f39439ee2c363b346/.github/workflows/cross-version.yml#L33-L36) against multiple versions of [Spring Boot](https://github.com/spring-projects/spring-boot) to ensure backward compatibility.

Specifically, scordio/spring-batch-notion#15:
* [Fails with Spring Boot 3.0.0](https://github.com/scordio/spring-batch-notion/actions/runs/5874903972/job/15930377843), which [implicitly uses](https://github.com/spring-projects/spring-boot/blob/v3.0.0/gradle.properties#L7) Kotlin 1.7.21
* [Fails with Spring Boot 3.0.9](https://github.com/scordio/spring-batch-notion/actions/runs/5874903972/job/15930377968), which [implicitly uses](https://github.com/spring-projects/spring-boot/blob/v3.0.9/gradle.properties#L7) Kotlin 1.7.21
* [Works with Spring Boot 3.1.0](https://github.com/scordio/spring-batch-notion/actions/runs/5874903972/job/15930378056), which [implicitly uses](https://github.com/spring-projects/spring-boot/blob/v3.1.0/gradle.properties#L7) Kotlin 1.8.21
* [Works with Spring Boot 3.1.2](https://github.com/scordio/spring-batch-notion/actions/runs/5874903967/job/15930378256) (latest), which [implicitly uses](https://github.com/spring-projects/spring-boot/blob/v3.1.2/gradle.properties#L7) Kotlin 1.8.21

For example, with Spring Boot 3.0.0 / Kotlin 1.7.21, I get the following error when running a unit test that relies on certain `notion-sdk-core` types:

```
java.lang.NoClassDefFoundError: kotlin/enums/EnumEntriesKt

	at notion.api.v1.model.databases.query.sort.QuerySortDirection.<clinit>(QuerySortDirection.kt:7)
	at io.github.scordio.springframework.batch.extensions.notion.Sort$Direction.<clinit>(Sort.java:111)
	at io.github.scordio.springframework.batch.extensions.notion.Sort.<clinit>(Sort.java:33)
	at io.github.scordio.springframework.batch.extensions.notion.SortTests.testToString(SortTests.java:76)
		...
Caused by: java.lang.ClassNotFoundException: kotlin.enums.EnumEntriesKt
	at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:641)
	at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:188)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:520)
	... 88 more
```

Is there anything that could be done to keep backward compatibility at least with Kotlin 1.7?

--- a/pom.xml
+++ b/pom.xml
@@ -41,13 +41,15 @@
         <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
         <okhttp3.version>3.14.9</okhttp3.version>
         <okhttp4.version>4.11.0</okhttp4.version>
-        <okhttp5.version>5.0.0-alpha.11</okhttp5.version>
+        <okhttp5.version>5.0.0-alpha.12</okhttp5.version>
         <gson.version>2.10.1</gson.version>
         <slf4j.version>1.7.36</slf4j.version>
         <slf4j2.version>2.0.11</slf4j2.version>
         <logback.version>1.2.10</logback.version>
-        <!-- Lock the version: https://github.com/seratch/notion-sdk-jvm/issues/91 -->
-        <kotlin.version>1.7.22</kotlin.version>
+        <kotlin.version>1.9.22</kotlin.version>
+        <!-- Allow use by clients on earlier versions of Kotlin-->
+        <kotlin.compiler.apiVersion>1.6</kotlin.compiler.apiVersion>
+        <kotlin.compiler.languageVersion>1.6</kotlin.compiler.languageVersion>
         <kotlin.code.style>official</kotlin.code.style>
         <duplicate-finder-maven-plugin.version>2.0.1</duplicate-finder-maven-plugin.version>
         <maven-deploy-plugin.version>3.1.1</maven-deploy-plugin.version>

